### ISSUE TITLE
Provide ability to queue/playlist audiobooks

### ISSUE THREAD

**Issue by digitlength** [advplyr/audiobookshelf-app#416]
### Is your feature request related to a problem? Please describe.
The kids sometimes listen to audiobooks when going to sleep. They can choose one each. At the moment, after the first book is finished I have to manually start the second book.

### Describe the solution you'd like
It would be handy to have a feature where I could enqueue audiobooks, so that the next book automatically starts after the one before it in the queue, until the queue is empty, or until the sleep timer has finished.

**Comment by advplyr** [comment]
I'm making this the main thread for requests about the audio player queue. This is already implemented in the website. Similar to Youtube, Abs distinguishes between a playlist and a queue. 

A playlist is a user-specific list of books or podcast episodes. A queue is a list of media items waiting to played in the audio player. A queue is visible in the audio player and you can add/remove items from the queue.

For podcasts when you start listening to a podcast episode a queue is automatically populated based on the pub date. For playlists and collections pressing play will automatically populate a queue.

**Comment by fidoriel** [comment]
My request is similar, but not the same.

I would love if an audiobook finishes, that next audiobook of the series gets automatically played. I do not want to queue them, just a simple check, maybe on the client like:

IF audiobook finished 
    and queue empty
    and audiobook in a series:
        play next part of series

**Comment by JReming85** [comment]
I know you guys are busy and most likely have a lot higher priority items then this small QoL fix, but is there any update?

**Comment by patienttruth** [comment]
I just for ABS up and running. Loving it.

Came here to see about queues and playlists on Android. Both would be awesome.

I figured out how to add something to a playlist, but it's hard to get back to it and it didn't proceed to the next track from what I could tell.

I'm not really sure how to help, but I'd be happy to if there's something I can do. Maybe with documentation down the road?

**Comment by patienttruth** [comment]
> Ideally, it'd be nice to create a Podcast playlist where you can add entire Podcasts rather than specific Podcast Episodes. Then the player would automatically queue all podcast Episodes from oldest-to-newest (adding new episodes to the end of the queue as they download).

I agree. I'm excited to see ABS develop. I can make it work, but there is some lost functionality vs AntennaPod. The main benefit I'm getting is that I can listen from the PC and have the playback synchronized. (Local archiving is nice too).

**Comment by RyuDragoon** [comment]
Just commenting to say I'd also very much enjoy this QoL added.

**Comment by politas** [comment]
This is a potentially lifesaving feature, enabling uninterrupted podcast playback while driving. I would hope it has top priority for the team.

**Comment by ncleggs** [comment]
I have just dipped my toe into ABS and tried it for the first time while driving today. I would love for the next podcast in my playlist to just play once the previous one ends, would be a life saver. To be honest, this may me to move away from ABS to something else at least temporarily until that comes in.

**Comment by nirurin** [comment]
I was listening to a podcast series for the first time from abs and it got to the end of episode 1 and then just...stopped. it didn't automatically play the next episode. 

I thought this was a bug, as that is just terrible design, but looking here it seems that it's not only intended but the devs has said they have no plans to fix it??

What?? Who listens to podcasts and wants it to force-stop at the end of every single episode?

**Comment by unknownmaster038** [comment]
Plus one on this request, definitely makes it difficult to listen to Podcasts while driving without this.

**Comment by d92t8b1i** [comment]
Another up-vote from me as well. Love ABS in general and this feature would greatly enhance the mobile app experience.

**Comment by brhahlen** [comment]
...and yet another upvote for this...
I've created a Playlist where I add my podcasts in, but it stops playing after each episode, which is annoying to say the least.

Love the rest of ABS, this should be a main feature, IMHO.

**Comment by andonevris** [comment]
This is over 2 years old now and still not implemented.  With the sheer volume of issues for this app I fear it will get buried in the noise and I consider this essential functionality for any podcast app.  

This has already been implemented in the web player, over a year ago, and still nothing for the app.

It's not too cumbersome for audiobooks which don't end so frequently but it's a real problem for podcasts.  I have switched back to my old podcast app because not having this is a deal breaker for me.

**Comment by freeagentkd35** [comment]
I hope this gets implemented on the app soon. It's such an amazing project with great features, and I considered this more important than many others already available.

**Comment by muddro1** [comment]
+1 on app queue!

**Comment by FalxMuralis** [comment]
+1, would really like to have this feature, especially for podcasts.

**Comment by gitchap** [comment]
+1 Would love to see this! Some option to auto add new podcast episodes to the cue (and download them!) would be even better!

**Comment by DeXP** [comment]
+1. I'd love to see the feature as well. I'm listening while riding a bike, it's not handy to stop.

**Comment by oberer** [comment]
+1. Would like to replace iTunes but the lack of an podcast autoplay feature is holding me back.

**Comment by d92t8b1i** [comment]
upvote!
This would be very handy during shorter podcasts.
Some channels have 5 min or less podcasts and if I start one before driving, I currently have to pull over every few mins to start the next episode.

**Comment by jucor** [comment]
Would love this queue for podcasts on Android, too! Big fan of ABS, it's terrific, that's just the one feature I'm missing the most.
Thanks for considering!

**Comment by marcaurele** [comment]
> I'm making this the main thread for requests about the audio player queue. This is already implemented in the website. Similar to Youtube, Abs distinguishes between a playlist and a queue.
> 
> A playlist is a user-specific list of books or podcast episodes. A queue is a list of media items waiting to played in the audio player. A queue is visible in the audio player and you can add/remove items from the queue.
> 
> For podcasts when you start listening to a podcast episode a queue is automatically populated based on the pub date. For playlists and collections pressing play will automatically populate a queue.

@advplyr I think what people would love to have is the implementation of the player queue on the mobile app itself, not only on the web interface. That way anyone can create their _on the fly_ playlist which is called "player queue" on the web interface.

**Comment by jucor** [comment]
Spot on! I did not even know there was a player queue on the web interface,
as I listen almost exclusively on the app 😅

**Comment by cliny** [comment]
Would just love an option in the android app to auto play the next podcast episode

**Comment by cameronj86** [comment]
Can someone vibe code the feature and share the fork since it's such a low priority with the project?

**Comment by splott74** [comment]
Yeah, it's been years. Is there some reason we don't have this?  I've been using audiobookshelf for years, for books. I finally set up podcasts, and now I'm realizing it's inherently broken. The whole point of audio is that I can go about my life and I don't have to be futzing with my phone all the time... Except that apparently I do. If I want to use ABS for podcasts. On my phone.

**Comment by StudioNirin** [comment]
No idea. I asume the dev uses abs for podcasts and only listens to the newest release, and anyone who doesnt do things their way is ignored as a heretic.

**Comment by Alcatraz077** [comment]
PR #1810 does it, just needs approved to be merged.

**Comment by ADD-OCD** [comment]
This has been sitting open a while, so wanted to flag: I maintain an unofficial fork ([Audiobookshelf+](https://github.com/ADD-OCD/audiobookshelf-app)) that implements most of this — continuous playback through downloaded series/collections, reliable podcast playlist auto-advance (survives Doze/backgrounding), and a visible, reorderable "up next" queue you can add individual items or whole series/playlists to. Android-only for now. Might save some of you the wait, or at least be useful reference if this ever gets tackled upstream.

**Comment by gardenlazy** [comment]
> This has been sitting open a while, so wanted to flag: I maintain an unofficial fork ([Audiobookshelf+](https://github.com/ADD-OCD/audiobookshelf-app)) that implements most of this — continuous playback through downloaded series/collections, reliable podcast playlist auto-advance (survives Doze/backgrounding), and a visible, reorderable "up next" queue you can add individual items or whole series/playlists to. Android-only for now. Might save some of you the wait, or at least be useful reference if this ever gets tackled upstream.

I was excited to try your fork. The primary feature I'm looking for is to have podcasts autoplay the next latest podcast when the current playing podcast finishes. Correct me if I'm wrong but I could not get this to work in your app but maybe your app simply does the other things that are mentioned in this thread and not that specific one. 

I tried playing from a specific podcast episode play button and I tried playing from the main button at the top of the podcast. Neither would play the next episode when the current episode completed.

**Comment by ADD-OCD** [comment]
that's likely a scope thing rather than a bug — the auto-advance only kicks in when you start playback from an actual Playlist page (Playlists → your playlist → Play), not from a podcast's own episode list or its main play button. Those don't set up a queue at all, so there's nothing to advance into once the episode ends.
To test it properly: create a Playlist, add a few podcast episodes to it in order, then hit Play from the Playlist's own page (not the podcast page). It should auto-advance through the playlist when each episode finishes, even backgrounded.
If you do that and it's still not advancing, let me know — I'd want to see the app's Logs (menu → Logs, share icon to export) from around when an episode finished, since that'd point at a real bug rather than a misunderstanding on my end.

**Comment by ADD-OCD** [comment]
> > This has been sitting open a while, so wanted to flag: I maintain an unofficial fork ([Audiobookshelf+](https://github.com/ADD-OCD/audiobookshelf-app)) that implements most of this — continuous playback through downloaded series/collections, reliable podcast playlist auto-advance (survives Doze/backgrounding), and a visible, reorderable "up next" queue you can add individual items or whole series/playlists to. Android-only for now. Might save some of you the wait, or at least be useful reference if this ever gets tackled upstream.
> 
> I was excited to try your fork. The primary feature I'm looking for is to have podcasts autoplay the next latest podcast when the current playing podcast finishes. Correct me if I'm wrong but I could not get this to work in your app but maybe your app simply does the other things that are mentioned in this thread and not that specific one. 
> 
> I tried playing from a specific podcast episode play button and I tried playing from the main button at the top of the podcast. Neither would play the next episode when the current episode completed.

that's likely a scope thing rather than a bug — the auto-advance only kicks in when you start playback from an actual Playlist page (Playlists → your playlist → Play), not from a podcast's own episode list or its main play button. Those don't set up a queue at all, so there's nothing to advance into once the episode ends.
To test it properly: create a Playlist, add a few podcast episodes to it in order, then hit Play from the Playlist's own page (not the podcast page). It should auto-advance through the playlist when each episode finishes, even backgrounded.
If you do that and it's still not advancing, let me know — I'd want to see the app's Logs (menu → Logs, share icon to export) from around when an episode finished, since that'd point at a real bug rather than a misunderstanding on my end.

**Comment by ADD-OCD** [comment]
@gardenlazy — thanks for actually trying it and reporting back, that's exactly the kind of feedback that's useful. You were right, and it turned out to be more than a Playlist-vs-episode-list mixup: I found a real bug — the podcast page's Play button found the next unfinished episode, but never set up a queue at all, so it stopped after that one episode instead of continuing. Fixed it so playing from the podcast page now queues and auto-advances through your remaining unfinished episodes, same as it already does for playlists. Will be in the next release shortly.

Since this is fork-specific behavior, would you mind opening any further fork issues on the fork's own repo (https://github.com/ADD-OCD/audiobookshelf-app/issues) rather than here going forward? Just trying to keep this thread on-topic for the original ask rather than turning it into fork support. Appreciate you testing it either way!

**Comment by ADD-OCD** [comment]
@gardenlazy — the fix is out now: [v0.9.0](https://github.com/ADD-OCD/audiobookshelf-app/releases/tag/v0.14.1-beta-plus-v0.9.0). Podcasts should auto-advance through remaining unfinished episodes from the podcast page's Play button. Let me know how it goes (on the fork repo, per above, if you hit anything else).
